main.go

```diff
@@ -100,6 +101,15 @@ func main() {
 	)
 	saCache.Start()
 
+	if *region == "" {
+		ec2MetadataClient := ec2metadatawrapper.New(nil)
+
+		metadataRegion, err := ec2MetadataClient.Region()
+		if err == nil {
+			*region = metadataRegion
+		}
```

Comment:
Should we fail fast for error?

Reply:
at a minimum, I think we should print an INFO log message stating that we experienced an error contacting IMDS.

Reply:
My thinking was that for someone who runs the webhook on a environment that is not ec2, the call to `ec2MetadataClient.Region()` would fail but the webhook should still continue its execution.
So I guess the underlying question is, is `amazon-eks-pod-identity-webhook` ec2-only?

Reply:
People run webhook on desktop for testing at least. We provide a flag for case like this. But it is indeed backwards incompatible (users who used to run webhook on desktop successfully have to start passing in a flag). 

But I would argue always injecting the AWS_REGION env variable is the correct behavior in the long term. By not checking the error, we risk not setting that env variable. 

That's the trade-off we need to make (my personal vote is enforcing region to be non-empty and hence fail the webhook if the call to IMDS failed). 

Looking forwards to other's thoughts.

Reply:
Agree with @jqmichael. Better to error out here.

Reply:
makes sense, got it.